app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,160 @@
+package org.oppia.app.player.state
+
+import android.text.Spannable
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.databinding.library.baseAdapters.BR
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.item_selection_interaction_items.view.*
+import kotlinx.android.synthetic.main.multiple_choice_interaction_items.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.ItemSelectionInteractionItemsBinding
+import org.oppia.app.databinding.MultipleChoiceInteractionItemsBinding
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.StringList
+import org.oppia.app.player.state.itemviewmodel.CustomizationArgsInteractionViewModel
+import org.oppia.app.player.state.itemviewmodel.SelectionContentViewModel
+import org.oppia.app.player.state.listener.InteractionAnswerRetriever
+import org.oppia.util.parser.HtmlParser
+
+private const val VIEW_TYPE_MULTIPLE_CHOICE = 1
+private const val VIEW_TYPE_ITEM_SELECTION = 2
+private const val INTERACTION_ADAPTER_TAG = "Interaction Adapter"
+
+/** Adapter to bind the interactions to the [RecyclerView]. It handles MultipleChoiceInput and ItemSelectionInput interaction views. */
+class InteractionAdapter(
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
+  private val explorationId: String,
+  private val itemList: MutableList<SelectionContentViewModel>,
+  private val customizationArgs: CustomizationArgsInteractionViewModel
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), InteractionAnswerRetriever {
```

Comment:
Shouldn't the top-level view be the interaction answer retriever? I can understand why this adapter class needs to have a method to construct the interaction object, but the adapter shouldn't be accessed outside of the owner of the recycler view using it.

To make this a bit cleaner, I suggest using a custom view (e.g. maybe a FrameLayout) that inflates its own view to set up the RecyclerView, and that FrameLayout can implement InteractionAnswerRetriever and be bound via the state adapter. This 'SelectionInteractionView' would have party with other interaction views to keep the calling code similar (e.g. ``interactionView as InteractionAnswerRetriever`` will still work).

Reply:
Done this in duplicate PR #276 .